radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,318 @@
+/**
+ * @file RBattery.c
+ * @date March 29, 2022
+ * @author Isaac Poirier (iap992)
+ */
+
+#include <RBattery.h>
+#include <RI2C.h>
+#include <string.h>
+#include <RCommon.h>
+
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
+/**
+ * Conversion Constants for between ADC Values and Real Values
+ */
+/** Used in: (Volts = Constant * ADC Count) */
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE		((float) 0.008993)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE			((float) 0.005865)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE			((float) 0.004311)
+
+
+/** Used in: (mAmps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BATTERY_OUTPUT_CURRENT_MAG		((float) 14.662757)
+#define ADC_COUNT_TO_5V_BUS_CURRENT_DRAW			((float) 1.327547)
+#define ADC_COUNT_TO_3V3_BUS_CURRENT_DRAW			((float) 1.327547)
+
+#define BATTERY_CURRENT_DIRECTION_THRESHOLD			((int) 512)
+
+/** Used in: (Temp in Deg C = (Scale * ADC Count) - Shift) */
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT 		((float) 273.15)
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE 		((float) 0.372434)
+
+#define ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT 		((float) 238.57)
+#define ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE 		((float) 0.3976)
+
+
+/**
+ * Constant Values for I2C Communication
+ */
+/** I2C Slave Address for EPS */
+#define BATTERY_I2C_SLAVE_ADDR 			(0x2A)
+
+/** Data Sent to the EPS via I2C for telemetry commands is 3 Bytes, 1 for the command, 2 for data*/
+#define BATTERY_TELEM_COMMAND_LENGTH	(3)
+
+/** Data sent to the EPS via I2C for all other commands is 2 Bytes, 1 for the command, 1 for data*/
+#define BATTERY_COMMAND_LENGTH			(2)
+
+/** Data Returned from the EPS via I2C is 4 Bytes */
+#define BATTERY_RESPONSE_LENGTH			(2)
+
+/** Delay between telemetry read/write operations with the EPS on I2C is 5ms */
+#define BATTERY_I2C_TELEM_DELAY			(5)
+
+/** Delay between other read/write operations with the EPS on I2C is 1ms */
+#define BATTERY_I2C_DELAY				(1)
+
+/** Number of voltage calls to the Battery */
+#define NUMBER_OF_VOLTAGE_COMMANDS (3)
+
+/** Number of current calls to the Battery*/
+#define NUMBER_OF_CURRENT_COMMANDS (4)
+
+/** Number of temperature calls to the Battery */
+#define NUMBER_OF_TEMP_COMMANDS (4)
+
+/** Filter Mask and Comparison value for Recognizing Telemetry Calls */
+#define TELEM_MASK 					(0x0000FF0000)
+#define TELEM_CALL_COMPARISON		(0x0000100000)
+
+
+/***************************************************************************************************
+                                          PRIVATE GLOBALS
+***************************************************************************************************/
+/** Charging or discharging enum for current direction of the battery */
+enum _batteryCurrentDirection {
+	discharging,
+	charging
+};
```

Comment:
Should make this static, and prefix enums with "battery" i.e. `batteryCharging` and `batteryDischarging`, according to style guide. Also the anonymous name (`_batteryCurrentDirection`) should be `_battery_current_direction`. You could also typedef the enum if you wanted